src/requests/consumption.js

```diff
@@ -0,0 +1,744 @@
+		let duplicateTasks = []
+
+		//taskMap = {
+		// 	projectTaskId : duplicateProjectTaskId
+		// }
+		let taskMap = {}
+
+		if (projectTemplates) {
+			projectTemplates.forEach((project) => {
+				project.externalId = project.externalId + common.SUFFIX_CHILD
+				delete project._id
+				project.updatedAt = new Date()
+				project.createdAt = new Date()
+				project.isReusable = false
+				templateProjectsTaskMap[project.externalId] = project.tasks
+				templateProjects.push(project)
+			})
+
+			Object.keys(templateProjectsTaskMap).forEach(async (projectExtId) => {
+				templateTaskIds = [...templateTaskIds, ...templateProjectsTaskMap[projectExtId]]
+			})
+			templateTaskIds = [...new Set(templateTaskIds)]
+			const projectsTaskCollection = mongoDb.collection(COLLECTIONS.TASKS)
+			const projectsTasksDetails = await projectsTaskCollection
+				.find({
+					_id: {
+						$in: templateTaskIds,
+					},
+				})
+				.toArray()
+
+			projectsTasksDetails.forEach((projectTask) => {
+				projectTask.externalId = utils.generateUniqueId()
+				taskMap[projectTask._id] = projectTask.externalId
+				projectTask.updatedAt = new Date()
+				projectTask.createdAt = new Date()
+				delete projectTask._id
+				duplicateTasks.push(projectTask)
+			})
+
+			await projectsTaskCollection.insertMany(duplicateTasks)
+
+			const projectsTasksDetailsAfterInsert = await projectsTaskCollection
+				.find({
+					externalId: {
+						$in: duplicateTasks.map((tasks) => tasks.externalId),
+					},
+				})
+				.toArray()
+
+			taskMap = _.mapValues(taskMap, (externalId) => {
+				// Find the corresponding object from projectsTasksDetailsAfterInsert
+				const task = _.find(projectsTasksDetailsAfterInsert, { externalId: externalId })
+
+				// If found, replace externalId with _id; otherwise, keep the externalId
+				return task ? ObjectId(task._id) : externalId
+			})
+
+			templateProjects.forEach((project) => {
+				let projectTasks = []
+				project.tasks.forEach((task) => {
+					projectTasks.push(taskMap[task])
+				})
+				project.tasks = projectTasks
+			})
+
+			await projectsCollection.insertMany(templateProjects)
+		}
+
+		const projectTemplatesAfterInsert = await projectsCollection
+			.find({
+				externalId: {
+					$in: templateProjects.map((projects) => projects.externalId),
+				},
+			})
+			.toArray()
+
+		// Add a new 'type' key to each project
+		const updatedProjectTemplates = projectTemplatesAfterInsert.map((project) => ({
+			...project, // Spread the existing project fields
+			type: common.PROJECT,
+		}))
+
+		return [...updatedProjectTemplates]
+	}
+}
+
+/**
+ * Format Program Template
+ * @name formatProgramTemplate
+ * @param {Object} templateData - Program template data
+ * @returns {Object} - Response contains formatted template
+ */
+const formatProgramTemplate = (templateData) => {
+	try {
+		let language = templateData?.language
+			? templateData?.language
+			: templateData?.resource
+			? templateData?.resource.flatMap((resource) => {
```

Comment:
@adithyadinesh0412 can we check language is an array and also each object contain label ?